source/common/common/macros.h

```diff
@@ -41,6 +41,15 @@ namespace Envoy {
     return *objectptr;                                                                             \
   } while (0)
 
+/**
+ * @brief Assert memory bounds to avoid copy errors.
+ */
+#define SAFE_MEMCPY(dst, src)                                                                      \
```

Comment:
Hi @jmarantz , I wanted to ask you if you would think appropiate to change this for the following template (as suggested in the issue #9328 )?
I appreciate any insight ;-).

```
diff --git a/source/common/common/macros.h b/source/common/common/macros.h
index 4be2134..da455e9 100644
--- a/source/common/common/macros.h
+++ b/source/common/common/macros.h
@@ -44,11 +44,13 @@ namespace Envoy {
 /**
  * @brief Assert memory bounds to avoid copy errors.
  */
-#define SAFE_MEMCPY(dst, src)                                                                      \
-  do {                                                                                             \
-    static_assert(sizeof(*(src)) == sizeof(*(dst)));                                               \
-    memmove(dst, src, sizeof(*(src)));                                                             \
-  } while (0)
+template<typename T1, typename T2>
+inline void safe_memcpy(T1 dst, T2 src) {
+
+  static_assert(sizeof(*(src)) == sizeof(*(dst)));
+  memmove(dst, src, sizeof(*(src)));
+
+}
```

Reply:
Yes however then I guess it shouldn't go into macros.h. Just give it its own header, safe_memcpy.h or something.

Also we should use memcpy rather than memmove. memmove can be more expensive and we could have a separate macro where that is required.